packages/lodestar/src/sync/initial/fast/index.ts

```diff
@@ -72,8 +73,10 @@ export class FastSync extends (EventEmitter as {new (): InitialSyncEventEmitter}
     this.targetCheckpoint = getCommonFinalizedCheckpoint(this.config, this.getPeerReputations());
     // head may not be on finalized chain so we start from finalized block
     // there are unfinalized blocks in db so we reprocess all of them
-    this.lastProcessedBlock = (await this.db.blockArchive.lastValue())!;
-    this.blockImportTarget = this.lastProcessedBlock.message.slot;
+    const lastKnownState = (await this.db.stateArchive.lastValue())?.clone() as TreeBacked<BeaconState>;
+    this.lastProcessedBlock = lastKnownState.latestBlockHeader.clone();
+    this.lastProcessedBlock.stateRoot = lastKnownState.hashTreeRoot();
```

Comment:
why do you need state root in sync anyways?

Reply:
without doing this we'll not get correct block root calculated from `this.lastProcessedBlock`
every time we process a block, we don't have a final `latestBlockHeader` as in https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar-beacon-state-transition/src/block/blockHeader.ts#L24 bc we don't have stateRoot at that time

Reply:
maybe just get finalized block root from forkChoice?

Reply:
we want both blockRoot and slot so `lastProcessedBlock` is being model as `BeaconBlockHeader`